tmt/result.py

```diff
@@ -310,9 +318,82 @@ def interpret_result(self, interpret: ResultInterpret) -> 'Result':
         :returns: :py:class:`Result` instance containing the updated result.
         """
 
-        if interpret in (ResultInterpret.RESPECT, ResultInterpret.CUSTOM):
+        if interpret == ResultInterpret.CUSTOM:
             return self
 
+        # Check for failed checks
+        checks_failed = any(check.result == ResultOutcome.FAIL for check in self.check)
+
+        if interpret == ResultInterpret.RESPECT:
+            if checks_failed:
+                self.result = ResultOutcome.FAIL
+                if self.note:
+                    self.note += ', check failed'
+                else:
+                    self.note = 'check failed'
+            return self
+
+        # Handle XFAIL for the overall test result
+        if interpret == ResultInterpret.XFAIL:
+            if self.result == ResultOutcome.PASS and checks_failed:
+                self.result = ResultOutcome.FAIL
+                if self.note:
+                    self.note += ', check failed'
+                else:
+                    self.note = 'check failed'
+            elif self.result == ResultOutcome.FAIL:
+                self.result = ResultOutcome.PASS
+                if self.note:
+                    self.note += ', expected failure'
+                else:
+                    self.note = 'expected failure'
+            elif self.result == ResultOutcome.PASS:
+                self.result = ResultOutcome.FAIL
+                if self.note:
+                    self.note += ', unexpected pass'
+                else:
+                    self.note = 'unexpected pass'
+            return self
+
+        # Handle individual check results
+        for check in self.check:
+            check_result = ResultInterpret(check.result)
```

Comment:
The field should already be `ResultInterpret`, as mentioned in a comment above WRT its definition. And I just noticed that `Test.result` isn't an enum, which is definitely incorrect, and a bad example :/